src/content/blog/web-framework-performance-report.mdx

```diff
@@ -0,0 +1,132 @@
+>
+> -- Anne Burnes, [RebelMouse](https://rebelmouse.com/)
+
+This will be an interesting metric to watch in 2023, and Google continues to weigh adding INP as an official Core Web Vital.
+
+# Lighthouse Performance
+
+[Lighthouse](https://developer.chrome.com/docs/lighthouse/overview/) is another tool that we can use to measure the user experience of a website. HTTP Archive runs Lighthouse in simulated mobile loading conditions. This offers a significantly more detailed and consistent analysis of page load performance, down to 100ms fractions of a second. Instead of looking at large “good” vs. “bad” thresholds and buckets, Lighthouse gives you a more detailed performance score measured out of 100.
+
+Real user data like Core Web Vitals is still the best measurement of the the real user experience, and you can see how the real experience vs. the lab experience differ in some of the charts below. However, there is still interesting insights to be learned from the extra detail that Lighthouse provides. Let's take a look at the data.
+
+<LighthouseMedianChart />
+
+In the interest of consistency, we have kept the original order from the previous section. However, you will notice that Remix appears much stronger on performance on Lighthouse than it did in the CWV Assessment. One explanation for this may be Remix's use of `startTransition` and `requestIdleCallback` to [defer React hydration on page load.](https://github.com/remix-run/remix/blob/4988875856ba4007d4e1e4219b117ce9c7012489/packages/remix-dev/config/defaults/entry.client.react.tsx) This would theoretically translate to better performance in some lab situations (like Lighthouse) at the expense of increased first-input delay in other, real-world situations.
+
+Unfortunately, the median Lighthouse performance score is low across the board. Half of the frameworks tested had a median performance considered "poor" (49 or below) while the other half had a median score that "needs Improvement" (50-89). No frameworks reached a "good" median score of 90+.
+
+Across all tracked websites, the median performance score was a 34/100. To that end, half of our tested frameworks (Astro, SvelteKit, and Remix) did come in above the internet average.
+
+<LighthousePercentileChart />
+
+By breaking the data down by percentile, we can start to see some slightly more encouraging numbers with Astro and SvelteKit reaching a score of 90+ in the p90 or p95 percentiles. However, the data clearly shows that all websites and frameworks (including Astro) still struggle to achieve good performance in real-world situations.
+
+# **The Cost of JavaScript**
+
+The last thing that we wanted to explore was the relationship between framework choice, performance, and total JavaScript payload size in real-world usage. Do the fastest frameworks tend to be the ones that also send the least amount of JavaScript to the client?
+
+<CostOfJSChart />
```

Comment:
Persistent labels would really help on this chart, rather than just ones that appear on hover